src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,50 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkLogin(user);
+        checkNullValues(user);
+        checkAge(user);
+        checkPassword(user);
+        return storageDao.add(user);
+    }
+
+    private void checkLogin(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("User with such login already exist");
+        }
+    }
+
+    private void checkAge(User user) {
+        if (user.getAge() < 0 || user.getAge() < MIN_AGE) {
```

Comment:
what is the difference between `user.getAge() < 0` and `user.getAge() < MIN_AGE`?